x/interchainstaking/keeper/keeper.go

```diff
@@ -294,12 +290,12 @@ func (k *Keeper) SetValidatorForZone(ctx sdk.Context, zone *types.Zone, data []b
 		}
 
 	} else {
-		if !val.Jailed && validator.IsJailed() {
-			// checking in case validator was tombstoned
-			if err := k.EmitSigningInfoQuery(ctx, zone.ConnectionId, zone.ChainId, validator); err != nil {
-				return err
-			}
```

Comment:
I see what you mean now. I don’t believe the condition in l298 can be true if val.Tombstoned is true. 

If tombstoned is true, val.Jailed is already true.

Reply:
Looks like you're reviewing my old code, right?

Reply:
Nope, the comment pertains to the new code. Reviewing on phone and clicked on the wrong line for the comment. Comment is accurate though.

Reply:
I have a line to check `val.Tombstoned` before [here](https://github.com/quicksilver-zone/quicksilver/blob/c33d55844041b0af4951dc76304274b1ceb7e8b3/x/interchainstaking/keeper/keeper.go#L293), so I believe the condition still good

Reply:
The condition is fine in and of itself, but I don’t see why the check is needed (and the previous logic was already working afaics).

As before, erroring here will cause the request to never be satisfied, and thus be repeated indefinitely (along with failing all icq responses in the same tx).

unless we encounter a case that cannot be valid, (tombstoned is a genuine valid state), we must always return nil.

Reply:
I see we should update the return, but I don't really understand about `but I don’t see why the check is needed `, can you make a `change suggestion` for ez understand?